lib/datadog/core/telemetry/event.rb

```diff
@@ -334,6 +334,33 @@ def payload
           end
         end
 
+        # Telemetry class for the 'logs' event
+        class Log < Base
+          LEVELS = {
+            error: 'ERROR',
+            debug: 'DEBUG',
+            warn: 'WARN',
+          }.freeze
+
+          def type
+            'logs'
+          end
+
+          def initialize(message:, level:)
+            super()
+            @message = message
+            @level = LEVELS.fetch(level)
```

Comment:
What do you think, does it make sense to throw an `ArgumentError` here instead of a `KeyError`?

```ruby
LEVELS.fetch(level) { |k| raise ArgumentError, "Invalid log level :#{k}"}
```